### PR TITLE
Penguins now waddle + waddle organ

### PR DESCRIPTION
**Pull request by MMMiracles**
:cl: MMMiracles
tweak: Penguins now waddle.
add: Penguins can now be butchered for the source of their waddling.
/:cl:

[why]: funny penguin waddle haha!

the fact this wasn't done already is a crime

### CODE REVIEW

File: code/modules/mob/living/simple_animal/friendly/penguin.dm [hunk 1]
```diff
@@ -17,6 +17,10 @@
 
 	do_footstep = TRUE
 
+/mob/living/simple_animal/pet/penguin/ComponentInitialize()
```
Comment (vuonojenmustaturska) [inline code/modules/mob/living/simple_animal/friendly/penguin.dm]: just use `Initialize`

**Review by vuonojenmustaturska — CHANGES_REQUESTED** [review]

File: code/modules/surgery/organs/ears.dm [hunk 2]
```diff
@@ -94,3 +94,20 @@
...
+	..()
+	if(istype(H))
+		to_chat(H, "<span class='notice'>You suddenly feel like you've lost your balance.</span>")
+		waddle = H.AddComponent(/datum/component/waddling)
+
+/obj/item/organ/ears/penguin/Remove(mob/living/carbon/human/H,  special = 0)
+	..()
+	if(istype(H))
+		to_chat(H, "<span class='notice'>Your sense of balance comes back to you.</span>")
+		waddle.RemoveComponent()
```
Comment (ninjanomnom) [inline code/modules/surgery/organs/ears.dm]: `QDEL_NULL(waddle)`

File: code/modules/surgery/organs/ears.dm [hunk 3]
```diff
@@ -94,3 +94,20 @@
...
+	var/datum/component/waddle
+
+/obj/item/organ/ears/penguin/Insert(mob/living/carbon/human/H, special = 0, drop_if_replaced = TRUE)
+	..()
+	if(istype(H))
+		to_chat(H, "<span class='notice'>You suddenly feel like you've lost your balance.</span>")
+		waddle = H.AddComponent(/datum/component/waddling)
+
+/obj/item/organ/ears/penguin/Remove(mob/living/carbon/human/H,  special = 0)
+	..()
```
Comment (ninjanomnom) [inline code/modules/surgery/organs/ears.dm]: `. = ..()`

File: code/modules/surgery/organs/ears.dm [hunk 4]
```diff
@@ -94,3 +94,20 @@
...
 		H.dna.species.mutant_bodyparts -= "ears"
 		H.update_body()
+
+/obj/item/organ/ears/penguin
+	name = "penguin ears"
+	desc = "The source of a penguin's happy feet."
+	var/datum/component/waddle
+
+/obj/item/organ/ears/penguin/Insert(mob/living/carbon/human/H, special = 0, drop_if_replaced = TRUE)
+	..()
```
Comment (ninjanomnom) [inline code/modules/surgery/organs/ears.dm]: `. = ..()`

File: code/modules/mob/living/simple_animal/friendly/penguin.dm [hunk 5]
```diff
@@ -14,9 +14,14 @@
 	speak_chance = 1
 	turns_per_move = 10
 	icon = 'icons/mob/penguins.dmi'
+	butcher_results = list(/obj/item/organ/ears/penguin = 1)
```
Comment (ninjanomnom) [inline code/modules/mob/living/simple_animal/friendly/penguin.dm]: No meat?

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

**Comment by PKPenguin321** [comment]
😨

File: code/modules/food_and_drinks/food/snacks/meat.dm [hunk 6]
```diff
@@ -318,6 +327,10 @@
 	name = "gondola steak"
 	tastes = list("meat" = 1, "tranquility" = 1)
 
+/obj/item/reagent_containers/food/snacks/meat/steak/penguin
+	name = "penguin steak"
+	tastes = list("a weird mix of beef and cod fish" = 1)
```
Comment (ninjanomnom) [inline code/modules/food_and_drinks/food/snacks/meat.dm]: Please use a taste value more like the other tastes, you could just make it have
```suggestion
	tastes = list("beef" = 1, "cod fish" = 1)
```

File: code/modules/food_and_drinks/food/snacks/meat.dm [hunk 7]
```diff
@@ -380,6 +393,11 @@
 	cooked_type = /obj/item/reagent_containers/food/snacks/meat/cutlet/gondola
 	tastes = list("meat" = 1, "tranquility" = 1)
 
+/obj/item/reagent_containers/food/snacks/meat/rawcutlet/penguin
+	name = "raw penguin cutlet"
+	cooked_type = /obj/item/reagent_containers/food/snacks/meat/cutlet/penguin
+	tastes = list("a weird mix of beef and cod fish" = 1)
```
Comment (ninjanomnom) [inline code/modules/food_and_drinks/food/snacks/meat.dm]: ^

File: code/modules/food_and_drinks/food/snacks/meat.dm [hunk 8]
```diff
@@ -419,3 +437,7 @@
 /obj/item/reagent_containers/food/snacks/meat/cutlet/gondola
 	name = "gondola cutlet"
 	tastes = list("meat" = 1, "tranquility" = 1)
+
+/obj/item/reagent_containers/food/snacks/meat/cutlet/penguin
+	name = "penguin cutlet"
+	tastes = list("a weird mix of beef and cod fish" = 1)
```
Comment (ninjanomnom) [inline code/modules/food_and_drinks/food/snacks/meat.dm]: ^

File: code/modules/food_and_drinks/food/snacks/meat.dm [hunk 9]
```diff
@@ -267,6 +267,15 @@
...
 	foodtype = RAW | MEAT
 
+/obj/item/reagent_containers/food/snacks/meat/slab/penguin
+	name = "penguin meat"
+	desc = "A slab of penguin meat."
+	list_reagents = list("nutriment" = 2, "cooking_oil" = 3)
+	cooked_type = /obj/item/reagent_containers/food/snacks/meat/steak/penguin
+	slice_path = /obj/item/reagent_containers/food/snacks/meat/rawcutlet/penguin
+	filling_color = "#B22222"
+	tastes = list("a weird mix of beef and cod fish" = 1)
```
Comment (ninjanomnom) [inline code/modules/food_and_drinks/food/snacks/meat.dm]: ^

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

**Review by ninjanomnom — APPROVED** [review]

**Pull request merged**